### PR TITLE
feat(useVibrate): new function

### LINKED ISSUE #516: useVibrate
Reactive hook to provide physical feedback with device vibration hardware using the Vibration API.

### PR DESCRIPTION
**Pull request by bharathvaj-ganesan**
Closes #516

### CODE REVIEW

**Comment by crutch12** [comment]
Don't you know: does it API supports gamepads? :)

**Comment by bharathvaj-ganesan** [comment]
> Don't you know: does it API support gamepads? :)

Pardon me. I don't get this question.

File: packages/core/useVibrate/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+```js
+import { useVibrate } from '@vueuse/core'
+
+const vibrating = ref(false)
+
+useVibrate(vibrating)
+
+// vibrating.value = true => vibration will start
+// vibrating.value = false => vibration will end
```
Comment (cawa-93) [inline packages/core/useVibrate/index.md:19]:
```suggestion
const startVibration = () => vibrating.value = true
const stopVibration = () => vibrating.value = false
const toggleVibration = () => vibrating.value = !vibrating.value
```

File: packages/core/useVibrate/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const vibrating = ref(false)
+
+const { pause, resume } = useVibrate(vibrating, {
+  pattern: [300, 100, 200, 100, 1000, 300],
+  loop: true,
+})
+
+// resume() => vibration will start
+// pause() => vibration will end
```
Comment (cawa-93) [inline packages/core/useVibrate/index.md:36]: For me is not clear what differentness between using `vibrating.value = true/false` and using `pause/resume` controls.

**Review by cawa-93 — COMMENTED** [review]
You may also be interested in reading the [discussion](https://github.com/vueuse/vueuse/discussions/518#discussion-3370085)

File: packages/core/useVibrate/index.md (hunk 2 above)
Comment (wheatjs) [reply packages/core/useVibrate/index.md:36]: This is probably something we should discuss in #518 as well so we can make recommendations on when to use play/pause vs controlling a value like `vibrating`

**Comment by antfu** [comment]
Close in favor of #1082

**Pull request closed without merging**